pkg/event-manager/drivers/docker_backend.go

```diff
@@ -95,32 +101,49 @@ func (d *dockerBackend) Deploy(ctx context.Context, driver *entities.Driver) err
 	io.Copy(os.Stdout, rc)
 
 	return utils.Backoff(time.Duration(d.DeployTimeout)*time.Second, func() error {
-		created, err := d.dockerClient.ContainerCreate(ctx, &container.Config{
+		config := &container.Config{
 			Image: driver.Image,
 			Env:   bindEnv(secrets),
 			Cmd:   buildArgs(driver.Config),
 			Labels: map[string]string{
 				labelEventDriverID: driver.ID,
 			},
-		}, nil, nil, "")
+		}
+
+		freePort, err := GetFreePort()
+		if err != nil {
+			return errors.Wrap(err, "error get free port when creating container")
+		}
+
+		hostConfig := &container.HostConfig{}
+
+		if driver.Expose {
+			config.ExposedPorts = nat.PortSet{
+				"80/tcp": struct{}{},
```

Comment:
To my understanding, all event driver will listen to port 80? This is not configurable to the event driver, am I right?

Reply:
Yes there seems no way to config port when create create a eventdriver